### PR TITLE
Enable llvm wasm target

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: .ci_support/linux_64_.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ c_compiler_version:
 c_stdlib:
 - sysroot
 c_stdlib_version:
-- '2.12'
+- '2.27'
```
[the remark this answers is not in the document]
Comment (Shaikh-Ubaid) [reply .ci_support/linux_64_.yaml:8]: Thanks, I will try it. Also, do you know if we need to explicitly install c_stdlib like:

```console
diff --git a/recipe/meta.yaml b/recipe/meta.yaml
index 0af9edb..635f7c4 100644
--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -18,6 +18,7 @@ requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
+    - {{ stdlib("c") }}
     - cmake =3.21.1
     - make   # [unix]
     - xeus =3.0.5       # [build_platform != target_platform]
```

[the remark this answers is not in the document]
Comment (Shaikh-Ubaid) [reply .ci_support/linux_64_.yaml]: >/home/conda/wasi-sdk-21.0/bin/clang: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by /home/conda/wasi-sdk-21.0/bin/clang)

Looks like using `2.28` doesn't seem to help.

**Comment by h-vetinari** [comment]
@conda-forge-admin, please rerender

**Comment by h-vetinari** [comment]
So I don't know this recipe, but from what I can tell, the stuff in `home/conda/wasi-sdk-21.0/bin/clang` gets downloaded during the build, rather building it in conda-forge. It is therefore not a surprise that it relies on newer symbols and doesn't know where to find them, because these builds are not (fully) relocatable, and cannot find our `libm.so.6`, which is not under `/lib64`, but [under](https://conda-metadata-app.streamlit.app/?q=conda-forge%2Fnoarch%2Fsysroot_linux-64-2.28-he073ed8_2.conda) `$PREFIX/x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6`:
```
+ WASI_SDK_PATH=/home/conda/wasi-sdk-21.0
+ echo /home/conda/wasi-sdk-21.0
+ /home/conda/wasi-sdk-21.0/bin/clang --version
/home/conda/wasi-sdk-21.0/bin/clang: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by /home/conda/wasi-sdk-21.0/bin/clang)
```

Again, without knowing the details of this recipe, it's very likely that the wasi-sdk should be built in conda-forge itself for use in this recipe.

**Comment by h-vetinari** [comment]
Ah, nevermind what I said, what we need is to change the image to `alma8` here (I managed to confuse myself between what we're compiling against, and how this recipe actually needs a new glibc for _running_ the build tools).

This isn't in widespread use yet, but we can try with a newer docker image.

**Comment by h-vetinari** [comment]
@conda-forge-admin, please rerender

File: recipe/crtlib_dir.patch [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
++#define WITH_LFORTRAN_BINARY_MODFILES
++
++#endif // LFORTRAN_CONFIG_H
+diff --git a/src/runtime/legacy/CMakeLists.txt b/src/runtime/legacy/CMakeLists.txt
+index 3795d800c..bebd8460f 100644
+--- a/src/runtime/legacy/CMakeLists.txt
++++ b/src/runtime/legacy/CMakeLists.txt
+@@ -1,5 +1,5 @@
+ set(SRC
+-    ../../../src/libasr/runtime/lfortran_intrinsics.c
```
Comment (Shaikh-Ubaid) [inline recipe/crtlib_dir.patch:50]: .

File: recipe/crtlib_dir.patch [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
++    ${CMAKE_CURRENT_SOURCE_DIR}/../../../src/libasr/runtime/lfortran_intrinsics.c
```
Comment (Shaikh-Ubaid) [inline recipe/crtlib_dir.patch:51]: .

**Pull request closed without merging**